### PR TITLE
HOCS-1804 - MPAM cases are no longer getting overall deadline date

### PR DESCRIPTION
**Pull request by hubert-raich**

### CODE REVIEW

**Comment by zuzak** [comment]
Can you amend the commit message to explain why you're making the change?

**Comment by hubert-raich** [comment]
That has messed up the commit history but I have put in more details and also renamed one of the variables to make is slightly less confusing.

Thanks

Hubert

**Comment by zuzak** [comment]
I've rebased those commits for you. Could you double check that I didn't lose any changes in doing so?

File: src/shared/common/components/stage-summary.jsx [hunk 1]
```diff
@@ -119,7 +119,7 @@ class StageSummary extends Component {
                                 {summary.additionalFields && summary.additionalFields.map(({ label, value }) => this.renderRow({ label, value }))}
                             </tbody>
                         </table>
-                        {summary.deadlinesEnabled && <table className='govuk-table margin-left--small'>
+                        {summary.deadlinesEnabled && summary.deadlines && <table className='govuk-table margin-left--small'>
```
Comment (MarkWrightBjss) [inline src/shared/common/components/stage-summary.jsx:122]: I think "summary.deadlines" already incorporates "summary.deadlinesEnabled" making "deadlinesEnabled" redundant here.

**Comment by MarkWrightBjss** [comment]
I prefer the new "stageDeadlineEnabled" name - much clearer.

**Comment by hubert-raich** [comment]
HOCS-1804 MPAM cases are no longer getting overall deadline date

Moving summaryDeadlineEnabled so that id does not impact the overall
dedlineEnabled check which is also used to display the overall Case Deadline.
'summaryDeadlineEnabled' should only hide the stage deadlines when false.

I've also added a check on stage-summary.jsx to make sure the heading 'Stage deadlines' is only visible 
when we have at least 1 stage deadline to display

File: src/shared/common/components/stage-summary.jsx (hunk 1 above)
Comment (hubert-raich) [reply src/shared/common/components/stage-summary.jsx]: Done:) good spot!

**Pull request merged**